config/confighttp/confighttp_test.go

```diff
@@ -336,15 +366,34 @@ func TestHTTPClientSettingWithAuthConfig(t *testing.T) {
 	}
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
-			client, err := test.settings.ToClient(test.host, componenttest.NewNopTelemetrySettings())
+			// Omit TracerProvider and MeterProvider in TelemetrySettings as otelhttp.Transport cannot be introspected
+			client, err := test.settings.ToClient(test.host, component.TelemetrySettings{Logger: zap.NewNop(), MetricsLevel: configtelemetry.LevelNone})
```

Comment:
use componenttest to create the TelemetrySettings?

Reply:
As noted in the comment the `otelhttp.Transport` doesn't export the `RoundTripper` that it wraps and so there would be no way to test the containment order beyond that transport if the `componenttest.NewNopTelemetrySettings()` were used.